checker/src/main/java/org/checkerframework/checker/optional/OptionalVisitor.java

```diff
@@ -124,30 +271,133 @@ public OptionalVisitor(BaseTypeChecker checker) {
+    optionalIntOf = TreeUtils.getMethod("java.util.OptionalInt", "of", 1, env);
+    optionalIntOrElse = TreeUtils.getMethod("java.util.OptionalInt", "orElse", 1, env);
+    optionalIntOrElseGet = TreeUtils.getMethod("java.util.OptionalInt", "orElseGet", 1, env);
+    optionalIntOrElseThrow =
+        TreeUtils.getMethodOrNull("java.util.OptionalInt", "orElseThrow", 0, env);
+    optionalIntOrElseThrowSupplier =
+        TreeUtils.getMethod("java.util.OptionalInt", "orElseThrow", 1, env);
+    optionalIntToString = TreeUtils.getMethod("java.util.OptionalInt", "toString", 0, env);
+
+    optionalLongEmpty = TreeUtils.getMethod("java.util.OptionalLong", "empty", 0, env);
+    optionalLongGetAsLong = TreeUtils.getMethod("java.util.OptionalLong", "getAsLong", 0, env);
+    optionalLongHashCode = TreeUtils.getMethod("java.util.OptionalLong", "hashCode", 0, env);
+    optionalLongIfPresent = TreeUtils.getMethod("java.util.OptionalLong", "ifPresent", 1, env);
+    optionalLongIfPresentOrElse =
+        TreeUtils.getMethodOrNull("java.util.OptionalLong", "ifPresentOrElse", 2, env);
+    optionalLongIsEmpty = TreeUtils.getMethodOrNull("java.util.OptionalLong", "isEmpty", 0, env);
+    optionalLongIsPresent = TreeUtils.getMethod("java.util.OptionalLong", "isPresent", 0, env);
+    optionalLongOf = TreeUtils.getMethod("java.util.OptionalLong", "of", 1, env);
+    optionalLongOrElse = TreeUtils.getMethod("java.util.OptionalLong", "orElse", 1, env);
+    optionalLongOrElseGet = TreeUtils.getMethod("java.util.OptionalLong", "orElseGet", 1, env);
+    optionalLongOrElseThrow =
+        TreeUtils.getMethodOrNull("java.util.OptionalLong", "orElseThrow", 0, env);
+    optionalLongOrElseThrowSupplier =
+        TreeUtils.getMethod("java.util.OptionalLong", "orElseThrow", 1, env);
+    optionalLongToString = TreeUtils.getMethod("java.util.OptionalLong", "toString", 0, env);
+
+    objectGetClass = TreeUtils.getMethod("java.lang.Object", "getClass", 0, env);
+
+    optionalCreators =
+        Arrays.asList(
+            optionalEmpty,
+            optionalOf,
+            optionalOfNullable,
+            optionalDoubleEmpty,
+            optionalDoubleOf,
+            optionalIntEmpty,
+            optionalIntOf,
+            optionalLongEmpty,
+            optionalLongOf);
+
+    // There are no propagators in OptionalDouble, OptionalInt, or OptionalLong.
     optionalPropagators =
         optionalOr == null
             ? Arrays.asList(optionalFilter, optionalFlatMap, optionalMap)
             : Arrays.asList(optionalFilter, optionalFlatMap, optionalMap, optionalOr);
-    // TODO: add these eliminators:
-    //   hashCode, ifPresent, ifPresentOrElse (Java 9+ only), isEmpty, isPresent, toString,
-    //   Object.getClass
+
+    // Iterating through this is inefficient.  I should find a better way.
+    // One idea: check the class that declares the method (using isOptionalType) and then the name
```

Comment:
I would call `isOptionalType` before searching this list.  The other option is to add a declaration annotation for optional eliminators.  That's what we normally do instead of hardcoding methods.

Reply:
I've done this here.  But I still need to add annotations for the Optional primitive classes.